corehq/apps/reports/standard/deployments.py

```diff
@@ -387,33 +386,49 @@ def total_records(self):
     @property
     def rows(self):
         if self.warehouse:
-            mobile_user_and_group_slugs = set(
-                # Cater for old ReportConfigs
-                self.request.GET.getlist('location_restricted_mobile_worker') +
-                self.request.GET.getlist(ExpandedMobileWorkerFilter.slug)
-            )
-            users = ExpandedMobileWorkerFilter.user_es_query(
-                self.domain,
-                mobile_user_and_group_slugs,
-                self.request.couch_user,
-            ).values_list('_id', flat=True)
+            user_ids = self.get_user_ids()
             sort_clause = self.get_sql_sort()
             rows = ApplicationStatusFact.objects.filter(
-                user_dim__user_id__in=users
+                user_dim__user_id__in=user_ids
             ).order_by(sort_clause).select_related('user_dim', 'app_dim')
             if self.selected_app_id:
                 rows = rows.filter(app_dim__application_id=self.selected_app_id)
             self._total_records = rows.count()
-            return self.process_users(rows)
+            first = self.pagination.start
+            last = first + self.pagination.count
+            return self.process_facts(rows[first:last])
         else:
             users = self.user_query().run()
             self._total_records = users.total
             return self.process_rows(users.hits)
 
+    def get_user_ids(self):
+        mobile_user_and_group_slugs = set(
+            # Cater for old ReportConfigs
+            self.request.GET.getlist('location_restricted_mobile_worker') +
+            self.request.GET.getlist(ExpandedMobileWorkerFilter.slug)
+        )
+        user_ids = ExpandedMobileWorkerFilter.user_es_query(
+            self.domain,
+            mobile_user_and_group_slugs,
+            self.request.couch_user,
+        ).values_list('_id', flat=True)
+        return user_ids
+
     @property
     def get_all_rows(self):
-        users = self.user_query(False).scroll()
-        return self.process_rows(users, True)
+        if self.warehouse:
+            user_ids = self.get_user_ids()
+            rows = ApplicationStatusFact.objects.filter(
+                user_dim__user_id__in=user_ids
+            ).select_related('user_dim', 'app_dim')
+            if self.selected_app_id:
+                rows = rows.filter(app_dim__application_id=self.selected_app_id)
+            self._total_records = rows.count()
+            return self.process_facts(chunked(rows, 10000))
```

Comment:
what's the purpose of using chunked here?

Reply:
Er. I fixed it to use paginator to split larger query into multiple smaller queries.